evaluator/src/test/kotlin/com/xebia/functional/xef/evaluator/SuiteBuilderSpec.kt

```diff
@@ -0,0 +1,205 @@
+          contextDescription { simpleString.next() }
+          outputResponse { simpleString.next() }
+        }
+
+        itemSpec(simpleString.next()) {
+          contextDescription { simpleString.next() }
+          outputResponse { simpleString.next() }
+        }
+
+        itemSpec(simpleString.next()) {
+          contextDescription { simpleString.next() }
+          outputResponse { simpleString.next() }
+        }
+      }
+      .shouldBeLeft() shouldBe
+      listOf(
+        EmptySuiteSpecOutputDescription(0),
+        EmptySuiteSpecOutputDescription(1),
+        InvalidNumberOfItemSpecOutputResponse(0),
+        InvalidNumberOfItemSpecOutputResponse(1),
+        InvalidNumberOfItemSpecOutputResponse(2)
+      )
+  }
+
+  @Test
+  fun shouldInvalidWithOneItemSpecWithoutOneOutPutResponse() = runTest {
+    SuiteSpec(simpleString.next()) {
+        outputDescription { simpleString.next() }
+        outputDescription { simpleString.next() }
+
+        itemSpec(simpleString.next()) {
+          contextDescription { simpleString.next() }
+          outputResponse { simpleString.next() }
+        }
+
+        itemSpec(simpleString.next()) {
+          contextDescription { simpleString.next() }
+          outputResponse { simpleString.next() }
+          outputResponse { simpleString.next() }
+        }
+      }
+      .shouldBeLeft() shouldBe listOf<ValidationError>(InvalidNumberOfItemSpecOutputResponse(0))
+  }
+
+  @Test
+  fun shouldInvalidWithTwoItemsSpecWithoutOneOutputResponse() = runTest {
+    SuiteSpec(simpleString.next()) {
+        outputDescription { simpleString.next() }
+        outputDescription { simpleString.next() }
+
+        itemSpec(simpleString.next()) {
+          contextDescription { simpleString.next() }
+          outputResponse { simpleString.next() }
+        }
+
+        itemSpec(simpleString.next()) {
+          contextDescription { simpleString.next() }
+          outputResponse { simpleString.next() }
+        }
+      }
+      .shouldBeLeft() shouldBe
+      listOf<ValidationError>(
+        InvalidNumberOfItemSpecOutputResponse(0),
+        InvalidNumberOfItemSpecOutputResponse(1),
+      )
+  }
+
+  @Test
+  fun shouldInvalidWithOneItemSpecWithEmptyOutputResponse() = runTest {
+    val emptyOutputResponse = emptyString.next()
+
+    SuiteSpec(simpleString.next()) {
+        outputDescription { simpleString.next() }
+        outputDescription { simpleString.next() }
+
+        itemSpec(simpleString.next()) {
+          contextDescription { simpleString.next() }
+          outputResponse { simpleString.next() }
+          outputResponse { simpleString.next() }
+        }
+
+        itemSpec(simpleString.next()) {
+          contextDescription { simpleString.next() }
+          outputResponse { simpleString.next() }
+          outputResponse { emptyOutputResponse }
+        }
+      }
+      .shouldBeLeft() shouldBe listOf<ValidationError>(EmptyItemSpecOutputResponse(1))
```

Comment:
Same as above. I prefer the way `ItemSpecBuilderSpec` defines the data, but here we can also group some tests in a single one.